L/Libtask/bundled/task.c

```diff
@@ -0,0 +1,91 @@
+/*
+ task.c
+ task copying (aka continuation) for lightweight processes (symmetric coroutines)
+ */
+
+#include "julia.h"
+
+// Workaround for JuliaLang/julia#32812
+jl_task_t *vanilla_get_current_task(void)
+{
+    jl_ptls_t ptls = jl_get_ptls_states();
+    return (jl_task_t*)ptls->current_task;
```

Comment:
So I don't really understand https://github.com/JuliaLang/julia/pull/32812; so the following may be stupid, please forgive me: I am guessing that for some reason there is a problem with directly using `ccall` on `jl_get_current_task`, which is why you re-implement it here. Not sure what the problem is, though: Is it an inefficiency? Or wrong behaviour? Or what? (It might be cool to document this here in a quick comment?)

At the same time, this specific function is a source of ABI incompatibility. If you instead did the following, this function would work fine with all Julia versions from 1.3-1.6, regardless of which Julia version was used to produce the binary:
```suggestion
    extern JL_DLLEXPORT jl_value_t *jl_get_current_task(void);
    return jl_get_current_task();
```
Of course if an inefficiency is the reason for this code, then my suggestion makes no sense.

Reply:
Looking further, you dig far more into innards of the TLS and tasks anyway, so I guess this question is moot :-). Except that I still feel a comment explaining its existence beyond just pointing at an issue might be helpful, but I am just watching from the sidelines, so don't hesitate to ignore me on this :-)

Reply:
I just copied the file from Libtask.jl without looking at the concrete implementation :smile: 

It seems that the linked PR caused hangs on Julia 1.5: https://github.com/TuringLang/Libtask.jl/issues/59

I agree that the comment could/should be improved. I guess @KDr2 might be able to explain the details here?

Reply:
Sorry, I didn't dive deeply enough into Julia's internal when I wrote this. At that time I found that calling to `jl_get_current_task` using ccall didn't work anymore after I updated Julia, then I found the commit(https://github.com/JuliaLang/julia/pull/32812) which caused the issue by using git bisect, the newly added optimization (specialized codegen) for `jl_get_current_task` made it problematic (crash the process while calling it using call). So I add this new function to keep the old implementation and fix the problem.